### PR TITLE
feat: added method to show correct RP organization_name in OP pages

### PR DESCRIPTION
**Pull request by rglauco**
fix: Updated cryptography to 42.0.2 for rsa method

### CODE REVIEW

File: spid_cie_oidc/provider/views/__init__.py [hunk 1]
```diff
@@ -391,6 +392,18 @@ def attributes_names_to_release(self, request, session: OidcSession) -> dict:
             for i in filtered_user_claims.keys()
         ]
         return dict(
-            i18n_user_claims = i18n_user_claims,
-            filtered_user_claims = filtered_user_claims
+            i18n_user_claims=i18n_user_claims,
+            filtered_user_claims=filtered_user_claims
         )
+
+    def get_client_organisation_name(self, tc):
```
Comment (peppelinux) [inline spid_cie_oidc/provider/views/__init__.py:399]: organization

File: spid_cie_oidc/provider/views/__init__.py [hunk 2]
```diff
@@ -391,6 +392,18 @@ def attributes_names_to_release(self, request, session: OidcSession) -> dict:
...
+        fed_metadata = tc.metadata.get("federation_entity", {})
+        name = fed_metadata.get("organization_name", "")
+        if not name:
+            op_metadata = tc.metadata.get("openid_relying_party", {})
+            name = op_metadata.get("organization_name", "")
+            if not name:
+                name = op_metadata.get("client_name", "")
+                if not name:
+                    name = op_metadata.get("client_id", "")
```
Comment (peppelinux) [inline spid_cie_oidc/provider/views/__init__.py:408]:
```suggestion
        rp_metadata = (
            tc.metadata.get(
                "federation_entity", {}
            ).get("organization_name", "") or
            tc.metadata.get(
                "openid_relying_party", {}
            )
        )
        if rp_metadata:
            name = (
                rp_metadata.get("organization_name", "") or
                rp_metadata.get("client_name", "") or
                rp_metadata.get("client_id", "")
            )
```

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: spid_cie_oidc/provider/views/__init__.py (hunk 2 above)
Comment (rglauco) [reply spid_cie_oidc/provider/views/__init__.py]: accepted change, corrected a typo: 
rp_metadata = (
            tc.metadata.get(
                "federation_entity", {}
            ) ~.get("organization_name", "")~

File: spid_cie_oidc/provider/views/__init__.py [hunk 3]
```diff
@@ -391,6 +392,23 @@ def attributes_names_to_release(self, request, session: OidcSession) -> dict:
             for i in filtered_user_claims.keys()
         ]
         return dict(
-            i18n_user_claims = i18n_user_claims,
-            filtered_user_claims = filtered_user_claims
+            i18n_user_claims=i18n_user_claims,
+            filtered_user_claims=filtered_user_claims
         )
+
+    def get_client_organisation_name(self, tc):
```
Comment (peppelinux) [inline spid_cie_oidc/provider/views/__init__.py:399]:
```suggestion
    def get_client_organization_name(self, tc):
```

File: spid_cie_oidc/provider/views/authz_request_view.py [hunk 4]
```diff
@@ -198,10 +198,14 @@ def get(self, request, *args, **kwargs):
...
+        # context = {
+        #     "client_organization_name": tc.metadata.get(
+        #         "client_name", self.payload["client_id"]
+        #     ),
+
         context = {
-            "client_organization_name": tc.metadata.get(
-                "client_name", self.payload["client_id"]
-            ),
+            "client_organization_name": self.get_client_organisation_name(tc),
```
Comment (peppelinux) [inline spid_cie_oidc/provider/views/authz_request_view.py:208]:
```suggestion
            "client_organization_name": self.get_client_organization_name(tc),
```

File: spid_cie_oidc/provider/views/consent_page_view.py [hunk 5]
```diff
@@ -56,9 +56,7 @@ def get(self, request, *args, **kwargs):
         context = {
             "form": self.get_consent_form()(),
             "session": session,
-            "client_organization_name": tc.metadata.get(
-                "client_name", session.client_id
-            ),
+            "client_organization_name": self.get_client_organisation_name(tc),
```
Comment (peppelinux) [inline spid_cie_oidc/provider/views/consent_page_view.py:59]:
```suggestion
            "client_organization_name": self.get_client_organization_name(tc),
```

File: spid_cie_oidc/provider/views/authz_request_view.py [hunk 6]
```diff
@@ -198,10 +198,14 @@ def get(self, request, *args, **kwargs):
 
         # stores the authz request in a hidden field in the form
         form = self.get_login_form()()
+
+        # context = {
+        #     "client_organization_name": tc.metadata.get(
+        #         "client_name", self.payload["client_id"]
+        #     ),
+
```
Comment (peppelinux) [inline spid_cie_oidc/provider/views/authz_request_view.py:206]:
```suggestion

```

**Pull request merged**